### PR TITLE
ninth commit

### PR DESCRIPTION
**Pull request by AbhiTagline**
Question 9

### CODE REVIEW

**Comment by AbhiTagline** [comment]
I find the first version of this program more effective then the new that i have created now. This one "[0f53f9e]"

File: question9.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+night_interval = P.closed(datetime(2023, 1, 2, 0,00,1),datetime(2023, 1, 2,6,00,00))
+time_diff = datetime(2023, 1, 2,10,00,00)-datetime(2023, 1, 1, 10,00,00)
+date_list = [interval.lower]  # Start with the lower bound
+current_date = interval.lower 
+# to track the hours
+count = 0
+for i in range(int(time_diff.total_seconds()/3600)): # divided by 3600 to convert minutes to hours
+    current_date += timedelta(hours=1)
+    if current_date not in night_interval:
+        count += 1
```
Comment (himanshu-suthar-tagline) [inline question9.py]: I do think this is incorrect way of doing the question.

Please check the python 
1. datetime
2. datetime.datetime.time
3. timedelta
4. portion

module correctly.

File: question9.py [hunk 2]
```diff
@@ -1,19 +1,24 @@
...
-        date_list.append(current_date)
-print(f"The difference between the start time - {start_time} and end time - {end_time} is {count} Hours")
\ No newline at end of file
+
+start_date = datetime.datetime(2023, 1, 1, 10, 0, 0)
+end_date = datetime.datetime(2023, 1, 5, 10, 0, 0)
+main_interval = P.closed(start_date, end_date)
+night_interval_start = datetime.time(0, 0, 0)
+night_interval_end = datetime.time(6, 0, 0)
+night_interval = P.closed(night_interval_start, night_interval_end)
```
Comment (himanshu-suthar-tagline) [inline question9.py]: I think this is not used anywhere.

**Comment by himanshu-suthar-tagline** [comment]
Please add some lines between code and make code more readable. Currently it looks too dense.

**Comment by AbhiTagline** [comment]
> Please add some lines between code and make code more readable. Currently it looks too dense.

ok

**Pull request merged**